includes/multi-currency/MultiCurrency.php

```diff
@@ -544,6 +548,39 @@ public function update_single_currency_settings( string $currency_code, string $
 		}
 	}
 
+	/**
+	 * Updates the customer currencies option.
+	 *
+	 * @param int $order_id The order ID.
+	 *
+	 * @return void
+	 */
+	public function maybe_update_customer_currencies_option( $order_id ) {
+		$order = wc_get_order( $order_id );
+
+		if ( ! $order ) {
+			return;
+		}
+
+		$currency   = $order->get_currency();
+		$currencies = get_option( self::CUSTOMER_CURRENCIES_KEY );
+
+		// Skip if the currency is not a string or if the option is invalid.
+		if ( ! is_string( $currency ) || ! is_array( $currencies ) ) {
+			return;
```

Comment:
I'm thinking about the initial migration, let's assume I upgraded to the version that contains this change for the first time, I won't have the option set. `get_option` will return `false` here, right? So the customer currency won't be updated unless I go to the Analytics section to have it updated. What do you think about splitting the condition into two parts, one checking and bailing if the order currency isn't valid, and the second part resetting the currency option when the option doesn't exist, or isn't valid?

Reply:
Ok, missed the line in `register_customer_currencies` method on the MC class, the initialization will fill in that data. It was folded by GH, needed to unfold it :)

Reply:
Nope, it was on the `Analytics` initialization. So the question stays valid :)

Reply:
> So the customer currency won't be updated unless I go to the Analytics section to have it updated.

The migration will happen on any page of the admin dashboard if MultiCurrency is enabled, not necessarily on the Analytics page, and most probably will happen right after the plugin update process. This is the path that will lead to this behavior:

1. The Init method is called on the [MultiCurrency constructor](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/multi-currency/MultiCurrency.php#L231)
2. The Analytics class [gets instantiated](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/multi-currency/MultiCurrency.php#L272)
3. The init method is called on the [Analytics constructor](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/multi-currency/Analytics.php#L53)
4. The `register_customer_currencies` method is called [on the admin dashboard](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/multi-currency/Analytics.php#L64)

Reply:
That makes sense, thanks for the explanation! The customer currencies will be updated whenever the page loads with initializing the MultiCurrency class then. Gotcha.

Reply:
Oh, sorry, it is to be updated when the user is admin and has the rights to `manage_woocommerce`. Since we only use this data for the Analytics section, it won't matter, but when a buyer purchases something on the frontend, until the admin is logged in, the option will not get populated. And it will be a problem when someone tries to use that in the frontend on a fresh site. Has a low chance to happen though.

https://github.com/Automattic/woocommerce-payments/blob/5f75b8fd87ad8d34d350a7d7c0c4774fb60d6efe/includes/multi-currency/Analytics.php#L62-L65

Do you think we should cover all possibilities by replacing the `get_option` call with `self::get_all_customer_currencies` in https://github.com/Automattic/woocommerce-payments/blob/419ff2cbf599d1dd26eea966f9b18469f7a95a41/includes/multi-currency/MultiCurrency.php#L566 ?

Reply:
That makes sense. I updated the code on https://github.com/Automattic/woocommerce-payments/pull/7143/commits/b49072944d048424c7f34c236471096540fcfb3f. Thanks for the suggestion @tpaksu and @marcinbot!